api_views/books.py

```diff
@@ -22,30 +22,43 @@ def add_new_book():
     request_data = request.get_json()
     try:
         jsonschema.validate(request_data, add_book_schema)
-    except:
-        return Response(error_message_helper("Please provide a proper JSON body."), 400, mimetype="application/json")
-    resp = token_validator(request.headers.get('Authorization'))
+    except jsonschema.exceptions.ValidationError as exc:
+        return Response(error_message_helper(exc.message), 400, mimetype="application/json")
+    
+    auth_header = request.headers.get('Authorization')
+    resp = token_validator(auth_header)
+    
     if "expired" in resp:
-        return Response(error_message_helper(resp), 401, mimetype="application/json")
+        return Response(error_message_helper("Token expired. Please log in again."), 401, mimetype="application/json")
     elif "Invalid token" in resp:
-        return Response(error_message_helper(resp), 401, mimetype="application/json")
-    else:
-        user = User.query.filter_by(username=resp).first()
+        return Response(error_message_helper("Invalid token. Please log in again."), 401, mimetype="application/json")
+    
+    user = User.query.filter_by(username=resp).first()
+    if not user:
+        return Response(error_message_helper("User not found."), 404, mimetype="application/json")
 
-        # check if user already has this book title
-        book = Book.query.filter_by(user=user, book_title=request_data.get('book_title')).first()
-        if book:
-            return Response(error_message_helper("Book Already exists!"), 400, mimetype="application/json")
-        else:
-            newBook = Book(book_title=request_data.get('book_title'), secret_content=request_data.get('secret'),
-                           user_id=user.id)
-            db.session.add(newBook)
-            db.session.commit()
-            responseObject = {
-                'status': 'success',
-                'message': 'Book has been added.'
-            }
-            return Response(json.dumps(responseObject), 200, mimetype="application/json")
+    # Check if the user already has this book title
+    existing_book = Book.query.filter_by(user=user, book_title=request_data.get('book_title')).first()
+    if existing_book:
+        return Response(error_message_helper("Book already exists!"), 400, mimetype="application/json")
+    
+    # Add the new book
+    try:
+        new_book = Book(
+            book_title=request_data.get('book_title'),
+            secret_content=request_data.get('secret'),
+            user_id=user.id
+        )
+        db.session.add(new_book)
+        db.session.commit()
+        responseObject = {
+            'status': 'success',
+            'message': 'Book has been added.'
+        }
+        return Response(json.dumps(responseObject), 201, mimetype="application/json")
+    except Exception as e:
```

Comment:
Consider using a more specific exception type instead of catching generic Exception. This broad exception handling could mask important errors. Consider handling specific exceptions like SQLAlchemyError for database operations.